fbw-common/src/wasm/systems/systems/src/hydraulic/flap_slat.rs

```diff
@@ -666,472 +758,542 @@ mod tests {
+            })
+        }
+
+        fn flaps_angle(&self) -> Angle {
+            self.query(|a| a.flaps_slats.flap_surface_angle())
+        }
+
+        fn left_motor_speed(&self) -> AngularVelocity {
+            self.query(|a| a.flaps_slats.left_motor.speed())
+        }
+
+        fn right_motor_speed(&self) -> AngularVelocity {
+            self.query(|a| a.flaps_slats.right_motor.speed())
+        }
+
+        fn left_motor_flow(&self) -> VolumeRate {
+            self.query(|a| a.flaps_slats.left_motor.flow())
+        }
+
+        fn right_motor_flow(&self) -> VolumeRate {
+            self.query(|a| a.flaps_slats.right_motor.flow())
+        }
+
+        fn synchro_position(&self) -> Angle {
+            self.query(|a| a.flaps_slats.position_feedback())
+        }
+
+        fn max_synchro_position(&self) -> Angle {
+            self.query(|a| a.flaps_slats.max_synchro_gear_position)
+        }
+
+        fn flap_slat_speed(&self) -> AngularVelocity {
+            self.query(|a| a.flaps_slats.speed)
+        }
+
+        fn set_angle_request(mut self, position: Option<Angle>) -> Self {
+            self.command(|a| a.set_angle_request(position));
+            self
+        }
+
+        fn set_individual_angle_request(
+            mut self,
+            sfcc_one_position: Option<Angle>,
+            sfcc_two_position: Option<Angle>,
+        ) -> Self {
+            self.command(|a| a.set_angle_per_sfcc(sfcc_one_position, sfcc_two_position));
+            self
+        }
+
+        fn set_hyd_pressure(mut self, pressure: Pressure) -> Self {
+            self.command(|a| a.set_current_pressure(pressure, pressure));
+            self
+        }
+
+        fn set_individual_hyd_pressure(
+            mut self,
+            left_pressure: Pressure,
+            right_pressure: Pressure,
+        ) -> Self {
+            self.command(|a| a.set_current_pressure(left_pressure, right_pressure));
+            self
+        }
+
+        fn check_flaps_position(&mut self, expected_position: Angle) {
+            let expected_ppu = self.flap_angle_to_fppu(expected_position);
+            let expected_position_percent = self.query(|a| {
+                Ratio::new::<ratio>(
+                    interpolation(
+                        &a.flaps_slats.synchro_gear_breakpoints,
+                        &a.flaps_slats.final_surface_angle_carac,
+                        a.flaps_slats.position_feedback().get::<degree>(),
+                    ) / interpolation(
+                        &a.flaps_slats.synchro_gear_breakpoints,
+                        &a.flaps_slats.final_surface_angle_carac,
+                        a.flaps_slats.max_synchro_gear_position.get::<degree>(),
+                    ),
+                )
+            });
+
+            let position_percent: f64 = self.read_by_name("LEFT_FLAPS_1_POSITION_PERCENT");
+            assert_eq!(position_percent, expected_position_percent.get::<percent>());
```

Comment:
It's probably better to use `assert_about_eq` for floating point comparisons:
```suggestion
            let position_percent: f64 = self.read_by_name("LEFT_FLAPS_1_POSITION_PERCENT");
            assert_about_eq!(position_percent, expected_position_percent.get::<percent>());
```